### PR TITLE
fix(android): make getUri and isResource available for override

### PR DESCRIPTION
**Pull request by shubhamguptadream11**
## Summary:

Fixes following issues: 
- https://github.com/facebook/react-native/issues/46150
- https://github.com/facebook/react-native/issues/46155

## Changelog:

[ANDROID] [FIXED] - Make getUri and isResource open

### CODE REVIEW

**Comment by shubhamguptadream11** [comment]
Hi @cortinico,

Could you please take a moment to review this PR when you get a chance? Given its impact, it would be great if we could prioritise it.

Thank you!

**Comment by deepanshushuklad11** [comment]
@cortinico Please get this PR merged.

**Review by cortinico — APPROVED** [review]

File: packages/react-native/ReactAndroid/api/ReactAndroid.api [hunk 1]
```diff
@@ -6720,9 +6720,9 @@ public class com/facebook/react/views/imagehelper/ImageSource {
 	public final fun getSize ()D
 	public final fun getSource ()Ljava/lang/String;
 	public static final fun getTransparentBitmapImageSource (Landroid/content/Context;)Lcom/facebook/react/views/imagehelper/ImageSource;
-	public final fun getUri ()Landroid/net/Uri;
+	public fun getUri ()Landroid/net/Uri;
 	public fun hashCode ()I
-	public final fun isResource ()Z
+	public fun isResource ()Z
```
Comment (cortinico) [inline packages/react-native/ReactAndroid/api/ReactAndroid.api:6725]: Just a small heads up that this is not the right fix. This file is generated from the Kotlin code, so the code needs to be updated as well. I'll take care of it

Comment (deepanshushuklad11) [reply packages/react-native/ReactAndroid/api/ReactAndroid.api:6725]: @cortinico Can you also explain how this file gets generated for future reference ?

Comment (cortinico) [reply packages/react-native/ReactAndroid/api/ReactAndroid.api:6725]: This file gets generated with `./gradlew apiDump` (see https://github.com/Kotlin/binary-compatibility-validator)
Also we regenerate the file internally with Buck so even if you edit it, we'll always regenerate it once we import the PR

Comment (deepanshushuklad11) [reply packages/react-native/ReactAndroid/api/ReactAndroid.api:6725]: So ideally it should run after your [PR](https://github.com/facebook/react-native/pull/46093) .
But didn't happen .

Comment (bradfloodx) [reply packages/react-native/ReactAndroid/api/ReactAndroid.api:6725]: How about a "thank you" also there @deepanshushuklad11 

Thanks @cortinico

Comment (deepanshushuklad11) [reply packages/react-native/ReactAndroid/api/ReactAndroid.api:6725]: @bradfloodx Yes a big thanks to @cortinico.
Just wanted to clarify my doubts.

**Comment by shubhamguptadream11** [comment]
@cortinico I understand how the .api file is generated from the Kotlin code now. It looks like I need to mark both the uri and isResource properties as open in ImageSource.kt to ensure they are not marked as final in the .api file.

Here's the change I'm planning to make:

```
/** Get the URI for this image - can be either a parsed network URI or a resource URI. */
public open val uri: Uri = computeUri(context)

/** Get whether this image source represents an Android resource or a network URI. */
public open var isResource: Boolean = false
```
Would you like me to push this change, or are you planning to handle it?

**Pull request closed without merging**